src/otx/algo/hooks/recording_forward_hook.py

```diff
@@ -364,3 +366,56 @@ def func(
             saliency_maps = self._normalize_map(saliency_maps)
 
         return saliency_maps.reshape((batch_size, self._num_classes, height, width))
+
+
+class MaskRCNNRecordingForwardHook(BaseRecordingForwardHook):
+    """Dummy saliency map hook for Mask R-CNN model."""
+
+    @classmethod
+    def create_and_register_hook(cls) -> BaseRecordingForwardHook:
+        """Create this object and register it to the module forward hook."""
+        return cls()
+
+    def func(self, feature_map: torch.Tensor | Sequence[torch.Tensor], _: int = -1) -> None:
+        """Return None for saliency map hook to generate saliency maps from predictions."""
+        return
+
+    @classmethod
+    def get_sal_map_from_preds(cls, preds: list[InstanceSegBatchPredEntity], num_classes: int) -> torch.Tensor:
```

Comment:
In all hooks, `func` is used to generate saliency maps from some input. Should we also preserve this patterns for `MaskRCNNRecordingForwardHook`. The different would be that `func` will get as input predictions instead of the feature maps, but to some extend feature maps are also prediction of some part of the model - so it seem to follow the same logic. What do you think?

Reply:
Thank you, fixed